### PR TITLE
Initial implementation of the TestRendered event for reftests

### PR DESCRIPTION
**Pull request by jgraham**
This implements the changes detailed in https://github.com/web-platform-tests/rfcs/pull/34

### CODE REVIEW

**Comment by jugglinmike** [comment]
Would you mind adding some documentation for this feature?

**Comment by jgraham** [comment]
https://phabricator.services.mozilla.com/D52684 for the gecko changes and https://treeherder.mozilla.org/#/jobs?repo=try&revision=f9b130cf3d577ac8feed26408f4a91d253c7ca7d for a try push containing these changes and the gecko changes.

**Comment by jgraham** [comment]
> Would you mind adding some documentation for this feature?

I wouldn't. And indeed I have now added some.

**Review by gsnedders — APPROVED** [review]
(but I think I'd rather a further +1 here)

**Review by Ms2ger — APPROVED** [review]
+1 if you checked that the azure test failures are not relevant

File: docs/writing-tests/reftests.md [hunk 1]
```diff
@@ -96,16 +96,31 @@ leaf nodes.)
...
-fired, and web fonts (if any) are loaded. In some cases it is
-necessary to delay the screenshot later than this, for example because
-some DOM manipulation is required to set up the desired test
-conditions. To enable this, the test may have a `class="reftest-wait"`
-attribute specified on the root element. This will cause the
-screenshot to be delayed until the `load` event has fired and the
-`reftest-wait` class has been removed from the root element. Note that
-in neither case is exact timing of the screenshot guaranteed: it is
-only guaranteed to be after those events.
+By default reftest screenshots are taken after the follwoing
```
Comment (sideshowbarker) [inline docs/writing-tests/reftests.md]:
```suggestion
By default, reftest screenshots are taken after the following
```

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**